### ISSUE TITLE
Dew point, frost point calculations component

### ISSUE THREAD

**Issue by mgrouch** [esphome/feature-requests#2770]
**Describe the problem you have/What new integration you would like**

Dew point frost point calculations component needed (using pressure and relative humidity, or more accurate formulas see link)

https://en.wikipedia.org/wiki/Dew_point

**Please describe your use case for this integration and alternatives you've tried:**

**Additional context**

**Comment by nagyrobi** [comment]
Easy to be configured as template sensors:
https://esphome.io/cookbook/bme280_environment.html

**Comment by mgrouch** [comment]
Yes. It’s possible doing manually but why everyone would implement it and possibly introduce errors.

**Comment by nagyrobi** [comment]
What errors? Just copy and paste carefully.

**Comment by mgrouch** [comment]
There is also no frost point currently

**Comment by nagyrobi** [comment]
Alright. Please suggest the correct formulas for the calculations you desire.

**Comment by mgrouch** [comment]
> Alright. Please quote here the correct formulas for the calculations you desire. I'll be happy to check / implement.

I did. They are in the Wikipedia link I sent originally and I pasted them just above

**Comment by ThomDietrich** [comment]
I support this request. ESPHome does an excellent job providing just the right building blocks for comfortable use case driven configuration. Manually implementing an off-the-shelf formula like for the dew point feels like an odd exception.

Btw I took the formula from here: https://www.kevinahrendt.com/approximating-dew-points

**Comment by nagyrobi** [comment]
Some results. Pink line is first formula (ESPHome cookbook), light blue is the second, and dark blue is the third one.
All 3:
![image](https://github.com/esphome/feature-requests/assets/1550668/5db5524b-6d0e-4ad6-a49c-71ba0f0a6123)

1-to-1:
![image](https://github.com/esphome/feature-requests/assets/1550668/8b3d8eec-37af-4098-9927-d75c0c1bccdb)
![image](https://github.com/esphome/feature-requests/assets/1550668/d41b8f34-a695-4add-9deb-c8e3cc23dc91)
![image](https://github.com/esphome/feature-requests/assets/1550668/09314cb4-ed29-4b60-a07c-9ed51c055665)

Difference is negligible (certainly smaller than the precision of the big majority of the sensors supported). First formula (ESPHome cookbook) and the third one correlate best, I think the first one is very good for every day use. There's no error in the formula.

**Comment by ThomDietrich** [comment]
@nagyrobi great show of initiative! 🏆 I did not spend the time to compare the different methods and agree, all of this seems minor in difference. Thought on the third formula: This binary decision introduces an intermittent/discontinuous at 0°C. Maybe that is acceptable.

I personally would go with the second formula and reference the link from above in documentation for interested folks.
